compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -907,14 +907,21 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                     ])
                 });
 
-                // We must additionally check that the return type impls `Future`.
+                // We must additionally check that the return type impls `Future + Sized`.
```

Comment:
Note that we don't need to check this for the match arm above (`CoroutineClosure`) because they are sized by construction.

Reply:
This sentence could be nice as a comment there.